iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.hpp

```diff
@@ -140,11 +135,16 @@ class PortManager
     cxx::optional<RuntimeName_t>
     doesViolateCommunicationPolicy(const capro::ServiceDescription& service IOX_MAYBE_UNUSED) const noexcept;
 
+    void publishCurrentServiceRegistry() const noexcept;
+
   private:
     RouDiMemoryInterface* m_roudiMemoryInterface{nullptr};
     PortPool* m_portPool{nullptr};
     ServiceRegistry m_serviceRegistry;
     PortIntrospectionType m_portIntrospection;
+
+    /// @todo #415 can we avoid the nullptr here?
+    PublisherPortRouDiType::MemberType_t* m_serviceRegistryPublisherPortData{nullptr};
```

Comment:
In the long run I would prefer an additional wrapper object (to be named) which contains the Port, the `ServiceRegistry` and other required data. For example currently we lack a mutex for find and add/remove (was broken before with the map as well). This may or may not be required later (depends on final design), mainly whether we call find concurrently.

I prefer to group related information if possible. Currently there is no need to have the registry in shared memory though, but this is also an interesting option for the future.

TLDR: We should maybe consider extensibility and group the data.

Reply:
I agree. I will think about it makes sense to include it in this PR already.

Reply:
Wrapped the pointer in an `cxx::optional`, but decide against grouping things in `struct` for now. We can refactor that in future.